### ISSUE TITLE
Add support for proxy settings per collection

### ISSUE THREAD

**Issue by basovnik** [a85/POSTMan-Chrome-Extension#4739]
It would be nice to support proxy settings per collection. We might have only some collections which need proxy.

**Comment by numaanashraf** [comment]
Thanks for the suggestion @basovnik. We'll take this into consideration while we improve our support for proxies.

**Comment by kenlyon** [comment]
Yes, this would be handy. I'm working on writing a proxy, and only the requests related to that project would need my proxy settings. Switching the Global Proxy Configuration or off while switching between projects is a bit of a nuisance, and is easy to forget when the pressure is on!

**Comment by frank-adler** [comment]
We need this, too. 👍

**Comment by teunis90** [comment]
We need this too, missed this issue. I've requested the same thing in #7285 

In addition, as mentioned in #7285, this feature would be an alternative and more powerful implementation of:
- #3942
- #6380
- #6484

Hope this will be implemented soon!

**Comment by dresh** [comment]
yes, can we get this please?

**Comment by JonasMatos0** [comment]
Having proxy per collections would really be handful.
That does prevent automating tests through postman in scenarios in which external tokens are needed for local environment setup.

**Comment by michabbb** [comment]
2 years later... i would be happy to see this too, it makes no sense to configure a proxy only global....
actually, it needs to be set by collection and even by request to override it in case the call is special....
```
- global (like now)
 |-> collection (missing)
   |-> request (missing)
```

**Comment by bduteh** [comment]
Is there any progress on this issue? It is much needed

**Comment by herenickname** [comment]
I join, necessary feature

**Comment by teunis90** [comment]
I have the feeling this is not an easy feature. It would be appreciated if we at least get feedback if this feature is planned or won't/can't be implemented.

**Comment by chrs-myrs** [comment]
This would be highly useful.  Or if not this then maybe a whitelist of domains for which the proxy is only active.  We are using the proxy to get around IP whitelists, so it only applies to some collections.

**Comment by moehawamdeh** [comment]
This is a great feature that is missing from Postman, my suggestion is to also have it enabled in pre-request script. 

Or at least allow negated bypass in the global config
E.g
`!*.github.com `
To allow proxy on requests to Github domain

**Comment by mohammadmazen** [comment]
It's much needed issue, hope it will added soon!

**Comment by j5shi** [comment]
Can't believe it's still missing...need it so badly...

**Comment by mottersheadt** [comment]
Hey Postman team - this would be incredibly valuable. Please add support!

**Comment by SynapseScribe** [comment]
it s been 5 years and no progress on this basic feature.. what is happening?

**Comment by eprothro** [comment]
👍🏼 for our team as well.

An update from the Postman team or closing this issue would be helpful!

**Comment by fappaz** [comment]
Need this as well, though would prefer supporting proxy [per collection](https://github.com/postmanlabs/postman-app-support/issues/9060) as it's even more flexible.

**Comment by EasyThe** [comment]
Much needed feature. Would be great to use that same setting for when importing via URL of WSDL collections to Postman.

**Comment by anotherscotch** [comment]
Not only per collection, but **per environment** would be even more useful IMO. For cases when specific api hosts are white listed to specific IPs.
